### PR TITLE
Add page loading events on live patch

### LINKED ISSUE #2559: `phx:page-loading-stop` does not get triggered when `push_patch` is used
### Environment

* Elixir version (elixir -v): `Elixir 1.14.3 (compiled with Erlang/OTP 25)`
* Phoenix version (mix deps): `1.7.2`
* Phoenix LiveView version (mix deps): `0.18.18`
* Operating system:`macOS Ventura 13.2.1`
* Browsers you attempted to reproduce this bug on (the more the merrier): `Version 111.0.5563.110 (Official Build) (arm64)`
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: `Yes`

### Actual behavior

Phoenix JS `phx:page-loading-stop` event listener does not get triggered when `push_patch` is used, like it does with using `<.link patch />` -component.
Cause of this updating DOM on `phx:page-loading-stop` event listener can have different and unexpected behaviour when used together.

Single file app to reproduce - [bumpah/debug-phoenix-js](https://github.com/bumpah/debug-phoenix-js)

### Expected behavior

I would expect both of the methods `push_patch` and `<.link patch />` to work in similar way and trigger `phx:page-loading-stop`. On the [single file app](https://github.com/bumpah/debug-phoenix-js) I would expect `<i id="current-query-params" />` element to always have text content of the current url, whether text input or link is used to update url.

### PR DESCRIPTION
**Pull request by enewbury**
Fixes #2559 

## Previous Behavior
Using a `<.link` component triggers a `phx:page-loading-start` as soon as it is clicked and `phx:page-loading-stop` when the response comes back from the server, however if `push_patch` was used server side, no Dom events were fired.
## New behavior
As soon as the client receives a `live_patch` message on the channel, it fires a `phx:page-loading-start` and as soon as the browser history pushed to, emits a `phx:page-loading-stop`

## Testing
I was unable to find any relevant testing covering this area of the functionality, and I didn't feel familiar enough with the area to build out that scaffolding. But if any maintainer has ideas, I'd be glad to set something up.

## Discussion
This feels a bit less meaningful because the change is initiated on the server side, so the `start` event is basically fired when we already have the change. But it seems this is still the best we can do for giving the client as much info as we have.
I'm also not exactly sure how the interacts with the actual dom updating with the diff payloads. If i understand correctly, the diffing happens via separate messages, and based on my experimentation, seems to happen BEFORE the live_patch event comes in, so I'm guessing we don't really need to care about it. The client initiated patch with `<.link` also doesn't seem to take into account anything to do with the actual dom diffing from the new page.

Basically, I'm just not entirely sure we are measuring anything particularly valuable with this implementation. It's basically just the time it takes to call the browser history api, which should be negligible.

### CODE REVIEW

**Comment by chrismccord** [comment]
Thanks for the legwork on this! https://github.com/phoenixframework/phoenix_live_view/issues/2559#issuecomment-1635204880
